File: api/Models/Quiz.cs
using System.ComponentModel.DataAnnotations;

namespace api.Models;

public class Quiz
{
    public int QuizId { get; set; }

    [Required]
    [StringLength(100, MinimumLength = 3)]
    public string Title { get; set; } = string.Empty;

    [StringLength(500)]
    public string Description { get; set; } = string.Empty;

    // Navigation property
    public virtual List<Question> Questions { get; set; } = new List<Question>();
}

File: quiz-client/src/types/quiz.ts
export interface Option {
    optionId: number;
    text: string;
    isCorrect: boolean;
}

export interface Question {
    questionId: number;
    text: string;
    type: number;
    points: number;
    options: Option[];
}

export interface Quiz {
    quizId: number;
    title: string;
    description: string;
    questions: Question[];
}